### PR TITLE
Add guide for integrating new features with FP

### PR DESCRIPTION
**Pull request by clelland**
This addresses some of the concerns brought up in #69, and provides
additional guidance to spec authors looking to integrate with feature
policy.

### CODE REVIEW

**Comment by clelland** [comment]
@raymeskhoury -- does this make sense as the start of a guide for spec authors? Anything else you'd like to see mentioned?

File: integration.md [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+>     <html>
+>       <head>
+>         <script>document.write('<script src="important-security-feature.js"></scr'+'ipt>')</script>
+>         ...
+>
+> then an attacker could embed the site in a frame with the `document.write`
+> feature disabled, and bypass the critical security script.
+>
+> To avoid this, if disallowed in a feature-policy-aware document, calling
+> `document.write()` will fire a `SecurityError` exception on the `document`
```
Comment (raymeskhoury) [inline integration.md]: Interesting. I would be worried that people would start using feature policy without realizing this implication and open themselves up to security problems.

File: integration.md [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+> object. If disallowed in a feature-policy-oblivious document, calling
+> `document.write()` feature must cause the current page to be navigated to an
+> error page which provides the option to open the page that violated the policy
```
Comment (raymeskhoury) [inline integration.md]: This is good guidance. I think we will need to spec this though, at the point when legacy features actually start using this.

File: integration.md [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+> productivity measure,) and wanted to make this a policy-controlled feature,
+> then the developers would have a choice -- whether to define the new feature
+> as "block cat pictures", or as "load cat pictures".
+>
+> If the goal is to block cat pictures, such that they cannot be loaded into a
+> document at all, then defining the feature as "load cat pictures" makes
+> sense, and the feature would be given a default allowlist of `[*]`. This way,
+> cat pictures would be allowed by default on the entire web, but once disabled
+> in a document, would not be available anywhere in that document, even in
+> child frames.
```
Comment (raymeskhoury) [inline integration.md]: I think there are other reasons to express the feature in terms of the ability to do something:
-Double negatives are confusing and error-prone
-Consistency is a good thing to try to encourage across features.

**Comment by raymeskhoury** [comment]
Overall looks really good to me :+1:

File: integration.md (hunk 1 above)
Comment (clelland) [inline integration.md]: This is a hypothetical situation; one which we can avoid by more carefully defining the behaviour of the feature when disallowed. I've rewritten some of this to try to make that clearer.

File: integration.md (hunk 3 above)
Comment (clelland) [inline integration.md]: Added, thanks!

**Pull request merged**